src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -843,15 +861,24 @@ private static unsafe void ClearImpl(Array array, int index, int length)
             if (array == null)
                 throw new ArgumentNullException("array");
 
+#if !REAL_MULTIDIM_ARRAYS
+            // NOTE: ONCE WE GET RID OF THE IFDEFS, WE SHOULD RENAME THIS METHOD.
+            // Get the backing array if this is an MDArray instance
             array = array.FlattenedArray;
+#endif
 
             if (index < 0 || index > array.Length || length < 0 || length > array.Length)
                 throw new IndexOutOfRangeException();
             if (length > (array.Length - index))
                 throw new IndexOutOfRangeException();
 
+#if REAL_MULTIDIM_ARRAYS
+            // The above checks should have covered all the reasons why Clear would fail.
+            Debug.Assert(false);
```

Comment:
The comment seems out of place here. And why would this routine fail (i.e. Assert(false)) all the time in case of REAL_MULTIDIM_ARRAYS?